code/modules/reagents/chemistry/recipes/others.dm

```diff
@@ -233,12 +233,20 @@
 	name = "Soapification"
 	id = "soapification"
 	result = null
-	required_reagents = list("liquidgibs" = 10, "lye"  = 10) // requires two scooped gib tiles
+	required_reagents = list("cornoil" = 10, "lye" = 10)
 	min_temp = T0C + 100
 	result_amount = 1
-
+	
+/datum/chemical_reaction/soapification/gib
+	name = "Gib Soapification"
+	id = "gib_soapification"
+	required_reagents = list("liquidgibs" = 10, "lye"  = 10) // requires two scooped gib tiles
 
 /datum/chemical_reaction/soapification/on_reaction(datum/reagents/holder, created_volume)
+	var/location = get_turf(holder.my_atom)
+	new /obj/item/soap/nanotrasen(location)
+
+/datum/chemical_reaction/soapification/gib/on_reaction(datum/reagents/holder, created_volume)
 	var/location = get_turf(holder.my_atom)
 	new /obj/item/soap/homemade(location)
```

Comment:
you dont need 2 overrides, you could just have a var on the reaction thats like `soap_type` then new that.